systest/tests/equivocation_test.go

```diff
@@ -0,0 +1,110 @@
+package tests
+
+import (
+	"context"
+	"crypto/ed25519"
+	"sync"
+	"testing"
+
+	pb "github.com/spacemeshos/api/release/go/spacemesh/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/systest/chaos"
+	"github.com/spacemeshos/go-spacemesh/systest/cluster"
+	"github.com/spacemeshos/go-spacemesh/systest/testcontext"
+)
+
+func TestEquivocation(t *testing.T) {
+	t.Parallel()
+	const bootnodes = 2
+	cctx := testcontext.New(t, testcontext.Labels("sanity"))
+
+	keys := make([]ed25519.PrivateKey, cctx.ClusterSize-bootnodes)
+	honest := len(keys) / 2
+	if (len(keys)-honest)%2 != 0 {
+		honest--
+	}
+	for i := 0; i < honest; i++ {
+		_, priv, err := ed25519.GenerateKey(nil)
+		require.NoError(t, err)
+		keys[i] = priv
+	}
+	for i := honest; i < len(keys); i += 2 {
+		_, priv, err := ed25519.GenerateKey(nil)
+		require.NoError(t, err)
+		keys[i] = priv
+		keys[i+1] = priv
+	}
+	cctx.Log.Infow("fraction of nodes will have keys set up for equivocations",
+		zap.Int("honest", honest),
+		zap.Int("equivocators", len(keys)-honest),
```

Comment:
number of equivocators should be `(len(keys)-honest)/2`, as every 2 have the same key.